cpp/tests/interpolate/cubic_spline_test.cpp

```diff
@@ -115,3 +149,123 @@ TEST_F(CubicSplineTest, test_interpolate_full)
+    *indexes, cudf::test::fixed_width_column_wrapper<int>{{0, 0, 1, 1, 2, 2, 3, 3, 3}});
+}
+
+TEST_F(CubicSplineTest, test_parallel_search_middle_triple)
+{
+  cudf::test::fixed_width_column_wrapper<float> short_triple{
+    {0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> long_triple{
+    {0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 0.0, 0.5, 1.0, 1.5, 2.0,
+     2.5, 3.0, 3.5, 4.0, 0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0}};
+  cudf::test::fixed_width_column_wrapper<int> point_ids_column{
+    {0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5, 10, 15}};
+
+  auto indexes = cuspatial::detail::find_coefficient_indices(long_triple,
+                                                             point_ids_column,
+                                                             prefix_column,
+                                                             short_triple,
+                                                             rmm::cuda_stream_default,
+                                                             this->mr());
+
+  cudf::test::expect_columns_equivalent(
+    *indexes,
+    cudf::test::fixed_width_column_wrapper<int>{
+      {0, 0, 1, 1, 2, 2, 3, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 11}});
+}
+
+TEST_F(CubicSplineTest, test_parallel_search_single_end_values)
+{
+  cudf::test::fixed_width_column_wrapper<float> short_triple{{0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> long_triple{{4.0, 4.1, 4.5, 5.0, 10000.0}};
+  cudf::test::fixed_width_column_wrapper<int> point_ids_column{{0, 0, 0, 0, 0}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5}};
+
+  auto indexes = cuspatial::detail::find_coefficient_indices(long_triple,
+                                                             point_ids_column,
+                                                             prefix_column,
+                                                             short_triple,
+                                                             rmm::cuda_stream_default,
+                                                             this->mr());
+
+  cudf::test::expect_columns_equivalent(
+    *indexes, cudf::test::fixed_width_column_wrapper<int>{{3, 3, 3, 3, 3}});
+}
+
+TEST_F(CubicSplineTest, test_parallel_search_triple_end_values)
+{
+  cudf::test::fixed_width_column_wrapper<float> short_triple{
+    {0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 0, 1, 2, 3, 4}};
+  cudf::test::fixed_width_column_wrapper<float> long_triple{
+    {4.0, 4.1, 4.5, 5.0, 10000.0, 4.0, 4.1, 4.5, 5.0, 10000.0, 4.0, 4.1, 4.5, 5.0, 10000.0}};
+  cudf::test::fixed_width_column_wrapper<int> point_ids_column{
+    {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2}};
+  cudf::test::fixed_width_column_wrapper<int> prefix_column{{0, 5}};
+
+  auto indexes = cuspatial::detail::find_coefficient_indices(long_triple,
```

Comment:
Is it necessary to test detail APIs?

Reply:
It is a private function, but it had a bug, so I wrote tests for it.